### PR TITLE
completed task 7

### PR DESCRIPTION
**Pull request by Thomas-Smith0176**
completes POST /api/articles/article_id/comments endpoint

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -42,142 +42,196 @@ describe("GET /api/topics", () => {
...
 });
+
 describe("GET /api", () => {
   test("200: responds with an object describing all the available endpoints on your API", () => {
-    const endpointsContents = fs.readFile(`${__dirname}/../endpoints.json`)
-    return Promise.all([endpointsContents])
-      .then((endpointsContents) => {
-        const endpointsActual = JSON.parse(endpointsContents)
-        return request(app)
+    const endpointsContents = fs.readFile(`${__dirname}/../endpoints.json`);
```
Comment (daviidbartlett) [inline __tests__/app.test.js:48]: you can just `require` the file

File: __tests__/app.test.js [hunk 2]
```diff
@@ -42,142 +42,196 @@ describe("GET /api/topics", () => {
...
+    return Promise.all([endpointsContents]).then((endpointsContents) => {
```
Comment (daviidbartlett) [inline __tests__/app.test.js:49]: Why are you using `Promise.all` for one promise?

File: __tests__/app.test.js [hunk 3]
```diff
@@ -42,142 +42,196 @@ describe("GET /api/topics", () => {
...
+          Object.keys(response.body.endpoints).forEach((endpoint) => {
+            expect(response.body.endpoints[endpoint]).toMatchObject({
+              description: expect.any(String),
+              queries: expect.any(Array),
+              exampleResponse: expect.any(Object),
+            });
+          });
+          expect(Object.keys(response.body.endpoints).length).toBe(
+            Object.keys(endpointsActual).length
+          );
```
Comment (daviidbartlett) [inline __tests__/app.test.js:64]: If you left everything as a string, you'd have a nice time. Does this string look like that string?

File: __tests__/app.test.js [hunk 4]
```diff
@@ -42,142 +42,196 @@ describe("GET /api/topics", () => {
...
+    }
+    return request(app)
+      .post("/api/articles/banana/comments")
+      .send(newComment)
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe("Bad request");
+      });
+  });
+  test('400: responds with bad request when given a valid article_id with no corresponding article', () => {
```
Comment (daviidbartlett) [inline __tests__/app.test.js:224]: if there no corresponding article then it can't be found. 404?

File: controllers/comments-controllers.js [hunk 5]
```diff
@@ -15,4 +15,15 @@ exports.getComments = (req, res, next) => {
...
     })
     .catch(next);
+};
+
+exports.postComment = (req, res, next) => {
+    const { body, username } = req.body
+    const { article_id } = req.params
+    return insertComment(body, username, Number(article_id))
+    .then((comment) => {
+        console.log(comment)
```
Comment (daviidbartlett) [inline controllers/comments-controllers.js:25]: remove

File: models/comments-models.js [hunk 6]
```diff
@@ -9,4 +10,18 @@ exports.selectCommentsByArticleId = (articleId) => {
     .then((response) => {
         return response.rows
     });
-};
\ No newline at end of file
+};
+
+exports.insertComment = (body, username, article_id,) => {
+    const created_at = new Date(Date.now())
```
Comment (daviidbartlett) [inline models/comments-models.js:16]: does your DB not do this for you?

File: models/comments-models.js [hunk 7]
```diff
@@ -9,4 +10,18 @@ exports.selectCommentsByArticleId = (articleId) => {
...
+    const query = format(`
+    INSERT INTO comments
+    (body, author, article_id, votes, created_at)
+    VALUES
+    ( %L, %L, %s, 0, %L )
```
Comment (daviidbartlett) [inline models/comments-models.js:21]: again I'm not sure if you need to insert votes, I think it has a default value

File: models/comments-models.js (hunk 6 above)
Comment (Thomas-Smith0176) [reply models/comments-models.js:16]: I think it only does it for the initial seed data, as this is a new comment I though it'd have to be done in the model?

**Pull request merged**